tests/main/snap-session-agent-socket-activation/task.yaml

```diff
@@ -12,10 +12,16 @@ environment:
     USER_RUNTIME_DIR: /run/user/${TEST_UID}
 
 prepare: |
+    # Ensure that snapd.session-agent.socket is enabled.  This may not
+    # be the case on distributions where presets have been used to
+    # disable it.
+    systemctl --user --global enable snapd.session-agent.socket
```

Comment:
Please store a marker file and add a corresponding restore code to undo this.

Reply:
I think I've got this part implemented correctly now.  Things were a little confusing, since `is-enabled` was returning false on the Ubuntu targets where we'd enabled the socket via a symlink in `/usr/lib/systemd/user`.  It seems this command only considers symlinks in `/etc/systemd/user`.

This was also the cause of the ubuntu-core-16 test failure: systemctl thought the unit was disabled, so tried to create a new symlink to enable it, which failed due to `/etc/systemd/user` being read-only on that release.

Since hard shipping the `sockets.target.wants` symlink in the package seems to be a Debian-ism, I've removed it from the `data/systemd-user` Makefile and added it to the .deb packaging instead.  According to [Debian bug #764678](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=764678), debhelper 11.5 has support for enabling user units the standard way.  So we might be able to remove it from there too eventually.